specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
If the data size is 0 then the `SetValueIsSet()` is called, too, see e.g. https://github.com/Matroska-Org/libebml/blob/master/src/EbmlUInteger.cpp#L142

This is exactly what I would expect. If an unsigned integer (or signed) is present but has a data size of 0 it is handled the exact same way as if the data size was > 0 and the data portion equals 0. libebml will therefore treat (un)signed integers with data size 0 as being present. Matroska's default values do not come into play in such a case.

This is also how I've always understood the specs. A default value is used if and only if the element is not present in the file at all. The element's data size does not influence that decision.

I'm fine with allowing (un)signed integers with a data size of 0. It takes away some ambiguity; therefore it makes parsers a bit easier to write. Additionally it saves a byte in very rare cases (though the latter is more a nice side effect from my point of view and not a rationale for allowing them).